calcom/cal.com
# fix: [CAL-3880] remove i18n for all <Meta> or figure out how to SSR

## What does this PR do?




- Fixes https://github.com/calcom/cal.com/issues/15354
- Fixes [CAL-3880]



## Notes
I tried to keep it as isolated as possible (just affecting the components that have <HeadSeo /> tag) but I think this "pattern" of defaulting to english when there is no content yet is worth adding it straight to the `useLocale` hook because right now google is indexing pages with the keys of i18n elements (If there is an agreement to this I'm happy to move this defaultValue over to the `useLocale` component. In terms of weight/size I think it's nothing to worry about as it's only 162 KB, for reference, the signup page takes 4.2MB and part of that is downloading the `/en/common.json` so if there is an agreement to make `useLocale` default to english we could potentially avoid fetching `/en/common.json` when locale = 'en' resulting in a net zero change in terms of bundle for english users (and minimal for other users).

<img width="600" alt="Screenshot 2024-06-06 at 11 59 06 PM" src="https://github.com/calcom/cal.com/assets/70239736/9d742cba-f979-4495-83fe-f5f026e6ae1f">

<img width="279" alt="Screenshot 2024-06-06 at 11 54 13 PM" src="https://github.com/calcom/cal.com/assets/70239736/46437972-ac14-493c-b79d-3f8b6d776d07">

---

<img width="129" alt="Screenshot 2024-06-06 at 11 54 53 PM" src="https://github.com/calcom/cal.com/assets/70239736/a74fb2cf-fe1e-46e3-9ece-6a7711075607">


## Mandatory Tasks (DO NOT REMOVE)


## How should this be tested?



- You can test by disabling javascript and going to `/video/no-meeting-found`, right now you'll see  "no_meeting_found | Cal.com" with this PR you'll see "No Meeting Found | Cal.com". You can try this on any other page that use `<HeadSeo />` component.
